File: LabSoapJava/src/main/java/pl/kamilszustak/labsoapjava/configuration/SoapConfiguration.java
package pl.kamilszustak.labsoapjava.configuration;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.oxm.jaxb.Jaxb2Marshaller;
import pl.kamilszustak.labsoapjava.client.SoapConnector;

@Configuration
public class SoapConfiguration {
    @Bean
    public Jaxb2Marshaller getJaxb2Marshaller() {
        Jaxb2Marshaller marshaller = new Jaxb2Marshaller();
        marshaller.setContextPath("pl.kamilszustak.labsoapjava.schemas.api");

        return marshaller;
    }

    @Bean
    public SoapConnector getSoapConnector(Jaxb2Marshaller marshaller) {
        SoapConnector connector = new SoapConnector();
        connector.setDefaultUri("http://localhost:8080/soap");
        connector.setMarshaller(marshaller);
        connector.setUnmarshaller(marshaller);

        return connector;
    }
}

File: LabSoap/src/main/kotlin/pl/kamilszustak/labsoap/configuration/SoapConfiguration.kt
package pl.kamilszustak.labsoap.configuration

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.oxm.jaxb.Jaxb2Marshaller
import pl.kamilszustak.labsoap.client.SoapConnector

@Configuration
class SoapConfiguration {
    @Bean
    fun getJaxb2Marshaller(): Jaxb2Marshaller =
        Jaxb2Marshaller().apply {
            contextPath = "pl.kamilszustak.labsoap.schemas.api"
        }

    @Bean
    fun getSoapConnector(marshaller: Jaxb2Marshaller?): SoapConnector? =
        SoapConnector().apply {
            defaultUri = "http://localhost:8080/soap"
            setMarshaller(marshaller)
            unmarshaller = marshaller
        }
}